### PR TITLE
State Structure for Tests

### PR DESCRIPTION
**Pull request by nikeshnazareth**
This creates a state structure for tests, using the same intuition as #85, except it uses the latest version of the prover manager (which involves running tests against the `ETHProverManager` and `ERC20ProverManager`).

I'm mostly just copying and tweaking the tests from #85, and from the existing tests on main (in the *.outdated and *.suppressed files, which I will remove as they are consumed). It's incomplete but I thought it makes sense to push the current version to deprecate #85, and also so you can see the structure if desired.

In the process I've also made some improvements/bug fixes to the actual code:
- bug fix: prevent evicting a prover using a publication that was before their period
- bug fix: prevent evicting a vacant prover (they should use `claimProvingVacancy` instead)
- bug fix: prevent outbidding a zero bid (or active period) with another zero bid. If we ever get to a stage where a prover chooses a zero bid (either because they have an interest in the rollup, or because there is some external incentive), we shouldn't be able to kick them out by also bidding zero.
- bug fix: emit the original liveness bond in ProverEvicted rather than the slashed stake
- bug fix: include the total delayed publications in the checkpoint, so the number of delayed publications can be computed directly. I do not like passing intermediateCheckPoint to the verifier. I'd be open to a better suggestion. This was the only solution I could think of where:
   - the caller does not know the last proven checkpoint (the whole point of allowing the proof to start earlier is that the proven checkpoint might change after the transaction is created).
   - we do not save the total delayed publications with every checkpoint in the commitment store
  - Note that redefining the commitment to include the total delayed publication does not work because you can't retrieve the total delayed publication from the proven commitment (and we can't ask the caller to provide it because they don't know the last proven checkpoint)
- refactor: use the `scaleByPercentage` function (instead of `scaleBy`)
- comments: clarify that the `bid` function continues working on a period that has theoretically started but has not been triggered by a publication (so it's still the "next period")
- improvement? (I think, but let me know if I'm missing something): I took our sample configuration for `ProverManagerConfig` and actually made those default values so I could make the two prover managers concrete classes. Otherwise, there is no prover manager in our code base that can be deployed. I also added getters for the config parameters.
- improvement: There is now a `ProverVacancyClaimed` event (note that it is called from the constructor as well).

I should finish copying and expanding the test suites tomorrow.

## Summary by CodeRabbit

- **New Features**
  - Added external getter functions for all configuration parameters in the Prover Manager.
  - Introduced new events for vacancy claims and period transitions.
  - Prover Manager contracts for ETH and ERC20 are now deployable.
  - Added extensive test suites covering all period states, bidding, eviction, exit, and vacancy scenarios for both ETH and ERC20.
  - Added invariant and universal behavior tests.

- **Improvements**
  - Enhanced validation and error handling for bidding, eviction, and proof submission.
  - Improved temporal checks and fee scaling logic.
  - Simplified and clarified configuration management with default values.
  - Updated proof verification to include delayed publications and intermediate checkpoints.
  - Strengthened bid fee validation and temporal validations on publication timestamps.
  - Clarified auction continuation behavior and event emissions on vacancy claims.

- **Bug Fixes**
  - Fixed edge cases in period transitions and balance updates during bids, exits, and proofs.

- **Refactor**
  - Refactored mock contracts and test helpers for clarity and flexibility.
  - Updated interfaces and event names for consistency.
  - Simplified mock contracts by removing unnecessary state and functions.

- **Documentation**
  - Updated comments and clarifications in events and interfaces.

- **Tests**
  - Comprehensive new and refactored tests for all major contract behaviors and edge cases.

### CODE REVIEW

File: test/ProverManager/CurrentPeriodIsOpenTest.t.sol [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+    }
+
+    function test_CurrentPeriodIsOpen_evictProver_shouldNotRevertIfPublicationIsNotProven() public {
+        IInbox.PublicationHeader memory header;
+        header.timestamp = vm.getBlockTimestamp() - proverManager.livenessWindow() - 1;
+        header.id = checkpointTracker.LAST_PROVEN_ID() + 1;
+        vm.prank(evictor);
+        vm.expectRevert(BaseProverManager.PublicationAlreadyProven.selector, 0);
+        proverManager.evictProver(header);
+    }
```
[the remark this answers is not in the document]
Comment (nikeshnazareth) [reply test/ProverManager/CurrentPeriodIsOpenTest.t.sol]: This is incorrect. The syntax `vm.expectRevert(..., 0)` implies there are zero reverts with that specific error message. This test is not necessarily that evict prover will succeed, just that it will not revert on that particular check.

**Review by LeoPatOZ — APPROVED** [review]

File: test/signal/ETHBridge.t.sol.suppressed [hunk 2]
```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: MIT
```
Comment (LeoPatOZ) [inline test/signal/ETHBridge.t.sol.suppressed:1]: Why do we need this file along with the signal service suppressed one ?
